src/components/Api.js

```diff
@@ -0,0 +1,217 @@
+
+const obj = {name: "obbie", ocupation: "Crypto Daytrader"};
+
+class Api {
+  //refactoring needed
+  constructor({ baseUrl, headers }) {
+    this.baseUrl = baseUrl;
+    this.headers = headers;
+  }
+
+  test(){
+    const obj2 = {residance: "USA", hobby: "WallSteet Bets"}
+    obj2.__proto__ = obj
+
+    return obj2.name
+  }
+
+  deleteCard(cardId, handler) {
+    return fetch(`https://around-api.en.tripleten-services.com/v1/cards/${cardId}`, {
+      method: "DELETE",
+      headers: this.headers,
+    })
+      .then((response) => {
+        if (!response.ok) {
+          throw new Error(`HTTP error! Status: ${response.status}`);
+        }
+        console.log("Data deleted successfully");
+        return response
+      })
+
+      .catch((error) => {
+        console.error("Error deleting data:", error);
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`   after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.

Please, remove it from `Api` and add it in `index.js`.